apiclient/httpapiclient.go

```diff
@@ -0,0 +1,52 @@
+package apiclient
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+var instance *HttpApiClient
+
+type HttpApiClient struct {
+	client *http.Client
```

Comment:
I would recommend also adding the URL as an attribute. In real life applications we would have a dedicated HttpClient per service we need to communicate with, this is because the way we handle responses could be different depending on what service you call.

It's not a bad thing to have a generic one though, but it's more common to see dedicated clients per sevice